src/coreclr/pal/src/exception/remote-unwind.cpp

```diff
@@ -2213,6 +2213,57 @@ get_proc_name(unw_addr_space_t as, unw_word_t addr, char *bufp, size_t buf_len,
     return -UNW_ENOINFO;
 }
 
+#ifdef FEATURE_USE_SYSTEM_LIBUNWIND
+
+// Function typedef for unw_get_proc_info_in_range
+typedef int (*unw_get_proc_info_in_range_fn)(unw_word_t start_ip, unw_word_t end_ip, unw_word_t eh_frame_table, unw_word_t eh_frame_table_len, unw_word_t exidx_frame_table, unw_word_t exidx_frame_table_len, unw_addr_space_t as, unw_word_t ip, unw_proc_info_t *pi, int need_unwind_info, void *arg);
+
+// Static variables to cache the function pointer and the library handle
+static unw_get_proc_info_in_range_fn cached_unw_get_proc_info_in_range = NULL;
+static bool has_failed_to_load = false; // Track if loading has failed
+
+// Function to attempt to load unw_get_proc_info_in_range dynamically
+static bool
+TryGet_unw_get_proc_info_in_range()
+{
+    if (cached_unw_get_proc_info_in_range != NULL) return true;
+
+    // If we already failed to load before, do not retry
+    if (has_failed_to_load) return false;
+
+    void* handle = dlopen("libunwind.so.8", RTLD_LAZY);
+    if (!handle)
+    {
+        handle = dlopen("libunwind.so", RTLD_LAZY);
+        if (!handle)
+        {
+            has_failed_to_load = true;
+            return false;
+        }
+    }
+
+#define STRINGIFY(x) #x
+#define UPREFIX(fn) STRINGIFY(UNW_PASTE(UNW_PASTE(UNW_PASTE(_U,UNW_TARGET),_), fn))
+#define ULPREFIX(fn) STRINGIFY(UNW_PASTE(UNW_PASTE(UNW_PASTE(_UL,UNW_TARGET),_), fn))
+
+    cached_unw_get_proc_info_in_range = (unw_get_proc_info_in_range_fn)dlsym(handle, ULPREFIX(get_proc_info_in_range));
```

Comment:
Stuff like `_ULaarch64_get_proc_info_in_range_fn` or _U.. depending on if we built it local only or with remote capability. Exported symbols are a bit tricky to comprehend (https://github.com/libunwind/libunwind/issues/376), as logically this should be arch agnostic, but since there are (ABI breaking) historical reasons, we get arch in their names. Note that when building code with libunwind, then we use unw_ prefix, this is only for exported symbol.

Reply:
Okay. Sounds like this actually does make thing easier. We can leave it.